Jasmine/Common/ViewModels/LevelSelectorViewModel.swift

```diff
@@ -0,0 +1,94 @@
+import RealmSwift
+
+/// The ViewModel for the level selector part.
+class LevelSelectorViewModel: LevelSelectorViewModelProtocol {
+    private var realm: Realm
+    /// The Levels object that manages the level objects
+    private var levels: Levels
+    /// Default levels as a level array
+    private var rawDefaultLevels: [Level] {
+        return levels.original
+    }
+    /// Custom levels as a level array
+    private var rawCustomLevels: [Level] {
+        return levels.custom
+    }
+    /// The defaults levels in the game.
+    var defaultLevels: [GameInfo] {
+        return rawDefaultLevels.map { level in
+            GameInfo(uuid: level.uuid, levelName: level.name, gameType: level.gameType, gameMode: level.gameMode)
+        }
+    }
+    /// The custom levels in the game.
+    var customLevels: [GameInfo] {
+        return rawCustomLevels.map { level in
+            GameInfo(uuid: level.uuid, levelName: level.name, gameType: level.gameType, gameMode: level.gameMode)
+        }
+    }
+
+    init() {
+        guard let realm = try? Realm() else {
+            fatalError("Cannot create Realm")
+        }
+        self.realm = realm
+        levels = Levels(realm: realm)
+    }
+
+    /// Deletes the custom level
+    ///
+    /// - Parameter gameInfo: the game info
+    func deleteLevel(from gameInfo: GameInfo) {
+        let level = getLevel(from: gameInfo.uuid, inArray: rawCustomLevels)
+        guard (try? levels.deleteLevel(level)) != nil else {
+            assertionFailure("Cannot delete level, somehow")
+            return
+        }
+    }
+
+    /// Get the phrase explorer VM from the game info
+    ///
+    /// - Parameter gameInfo: the game info to be passed
+    func getPhraseExplorerViewModel(from gameInfo: GameInfo) -> PhrasesExplorerViewModel {
+        let level = getLevel(from: gameInfo.uuid, inArray: rawCustomLevels)
+        return PhrasesExplorerViewModel(phrases: level.phrases)
+    }
+
+    /// Play the game 
+    ///
+    /// - Parameter gameInfo: the game info to be passed
+    func playGame(from gameInfo: GameInfo) -> BaseViewModelProtocol {
```

Comment:
You can just `as?` it right? Since I'm returning the correct VM (`CiHuiSwappingViewModel` `TetrisGameViewModel` etc.)?